src/Lean/Linter/Basic.lean

```diff
@@ -5,18 +5,66 @@ Authors: Lars König
 -/
 prelude
 import Lean.Data.Options
+import Lean.MonadEnv
 import Lean.Log
 
 namespace Lean.Linter
 
+/-- Linter sets are represented as a map from linter name to set name,
+to make it easy to look up which sets to check for enabling a linter.
+-/
+def LinterSets := NameMap (Array Name)
+  deriving EmptyCollection, Inhabited
+
+def insertLinterSetEntry (map : LinterSets) (entry : Name × Array Name) : LinterSets :=
```

Comment:
Would you like to add a doc-string? For example, it could be helpful to indicate that `entry.1` is the name of the linter set and `entry.2` are the options contained in it.

Reply:
```suggestion
/-- Insert a set into a `LinterSets` map.

`entry.1` is the name of the linter set,
`entry.2` contains the name of the set's linter options.
-/
def insertLinterSetEntry (map : LinterSets) (entry : Name × Array Name) : LinterSets :=
```